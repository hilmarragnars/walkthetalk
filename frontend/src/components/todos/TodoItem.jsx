import React from 'react';
import { Button } from '@smooth-ui/core-sc';

function TodoItem(props) {

    const handleClick = () => {
      console.log(props.todo)
      props.onDelete(props.todo.id)
    }
    return (
      <li>
        <p>{props.todo.title}</p>
        <Button onClick={handleClick}>Delete</Button>
      </li>
    )
}

export default TodoItem
