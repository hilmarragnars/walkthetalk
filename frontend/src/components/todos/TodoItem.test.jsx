import TodoItem from './TodoItem';


const props = {
  todo: {
    id: 1,
    title: "Test todo items",
    description: "This is a short description about a todo item",
    completed: false,
  },
  onDelete: jest.fn()
}

describe('TodoItem rendering correctly', () => {
  it("Renders correctly empty", () => {
    const wrapper = shallow(
      <TodoItem todo={{}} onDelete={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly with a todo item", () => {
    const wrapper = shallow(
      <TodoItem todo={props.todo} onDelete={props.onDelete} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
