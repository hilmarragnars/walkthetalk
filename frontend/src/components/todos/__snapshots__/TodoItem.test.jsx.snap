// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem rendering correctly Renders correctly empty 1`] = `
<li>
  <p />
  <styled(Button)
    onClick={[Function]}
  >
    Delete
  </styled(Button)>
</li>
`;

exports[`TodoItem rendering correctly Renders correctly with a todo item 1`] = `
<li>
  <p>
    Test todo items
  </p>
  <styled(Button)
    onClick={[Function]}
  >
    Delete
  </styled(Button)>
</li>
`;
